test_scripts/Smoke/API/031_PerformAudioPassThru_PositiveCase.lua

```diff
@@ -0,0 +1,166 @@
+local function file_check(file_name)
+  local file_found = io.open(file_name, "r")
+  if nil == file_found then
+    return false
+  else
+    return true
+  end
+end
+
+local function expectOnHMIStatusWithAudioStateChanged_PerformAudioPassThru(self)
+  commonSmoke.SetAppType(config.application1.registerAppInterfaceParams.appHMIType)
+  local isMediaApplication = commonSmoke.GetAppMediaStatus()
+  if
+  true == isMediaApplication or
+  true == commonSmoke.HMITypeStatus.NAVIGATION then
+    self.mobileSession1:ExpectNotification("OnHMIStatus",
+      { hmiLevel = "FULL", audioStreamingState = "ATTENUATED", systemContext = "MAIN" },
+      { hmiLevel = "FULL", audioStreamingState = "ATTENUATED", systemContext = "HMI_OBSCURED" },
+      { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "HMI_OBSCURED" },
+      { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN" })
+    :Times(4)
+  elseif
+    false == isMediaApplication then
+      self.mobileSession1:ExpectNotification("OnHMIStatus",
+        { hmiLevel = "FULL", audioStreamingState = "NOT_AUDIBLE", systemContext = "HMI_OBSCURED" },
+        { hmiLevel = "FULL", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+      :Times(2)
+    end
+  end
+
+  local function sendOnSystemContext(self, pCtx, pAppID)
+    self.hmiConnection:SendNotification("UI.OnSystemContext",
+      { appID = pAppID, systemContext = pCtx })
+  end
+
+  local function performAudioPassThru(pParams, self)
+    local cid = self.mobileSession1:SendRPC("PerformAudioPassThru", pParams.requestParams)
+    pParams.requestUiParams.appID = commonSmoke.getHMIAppId()
+    EXPECT_HMICALL("TTS.Speak", pParams.requestTtsParams)
+    :Do(function(_,data)
+        self.hmiConnection:SendNotification("TTS.Started")
+        local function ttsSpeakResponse()
+          self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+          self.hmiConnection:SendNotification("TTS.Stopped")
+        end
+        RUN_AFTER(ttsSpeakResponse, 500)
+      end)
+    EXPECT_HMICALL("UI.PerformAudioPassThru", pParams.requestUiParams)
+    :Do(function(_,data)
+        sendOnSystemContext(self, "HMI_OBSCURED", pParams.requestUiParams.appID)
+        local function uiResponse()
+          self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+          sendOnSystemContext(self, "MAIN", pParams.requestUiParams.appID)
+        end
+        RUN_AFTER(uiResponse, 1500)
+      end)
+    expectOnHMIStatusWithAudioStateChanged_PerformAudioPassThru(self)
+    self.mobileSession1:ExpectNotification("OnAudioPassThru")
+    :Times(1)
+    :Timeout(11000)
+    self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+    :ValidIf (function()
+        local file = commonPreconditions:GetPathToSDL() .. "storage/" .. "audio.wav"
+        if file_check(file) ~= true then
+          return false, "Can not found file: audio.wav"
+        else
```

Comment:
@GetmanetsIrina Else statement can be omitted here.

Reply:
@aderiabin , Done in fb183773759e98c5603d803a4be087a44dba7ae6.